test/k8sT/Services.go

```diff
@@ -672,18 +672,22 @@ var _ = Describe("K8sServicesTest", func() {
 			), "Failed to account for IPv4 fragments to %s (out)", dstIP)
 		}
 
-		getIPv4Andv6AddrForIface := func(nodeName, iface string) (string, string) {
+		getIPv4AddrForIface := func(nodeName, iface string) string {
```

Comment:
I know you didn't modify the body of this function, but we might as well fix-up the incorrect expectation. Because the expectation is inside a function, we need to use `ExpectWithOffset`.

Reply:
We are using `res.ExpectSuccess` which internally uses `ExpectWithOffset`. I am not sure if I understand the usage of `ExpectWithOffset` correctly but do we still need to use that here?

Reply:
`ExpectWithOffset` is used when you define a helper function that's called inside an `It()` block. Ideally, all expectations should be done inside the `It()` block, and therefore using `Expect` is proper. However, sometimes it is convenient to define helpers and call them within `It()` blocks, hence the need for an "offset".

Reply:
In the case of `res.ExpectSuccess`, there are two offsets. One for the actual call to it, and one from the call to `getIPv4AddrForIface` from the `It()` block. It's confusing and unreadable, huh? :) And what's worse is that when a test fails with the wrong offset or `ExpectWithOffset` was wrongly used, the stacktrace for the CI failure is off, which makes debugging / triaging a pain. That's why I've been trying to simplify our "expectations" in Ginkgo.

Reply:
Ah, that makes sense. So for this, do you think creating a new helper function like `CmdRes.ExpectSuccessWithOffset` where we make use of right offset make sense? I don't want to edit the parts of the code the PR does not touch. Also, we can take this as a followup and I can fix all the `ExpectSuccess` calls from within a helper function using this new method.

Reply:
Hmm, I'm towards just "deprecating" `ExpectSuccess` because it's just as easy to do something like:

```
Expect(foo()).To(helper.CMDSuccess())
```
or
```
ExpectWithOffset(foo(), offset).To(helper.CMDSuccess())
```

So I don't really see the benefit for `ExpectSuccess` defined on the `CmdRes` object anyway, so I'd stay away from adding yet another helper function.

But for this PR, I agree. No need to make unrelated changes; can be a followup.